.github/workflows/tox-pytest.yml

```diff
@@ -0,0 +1,85 @@
+name: tox-pytest
+
+on: [push, pull_request]
+
+jobs:
+  ci-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+
+    - name: Set up conda environment for testing
+      uses: conda-incubator/setup-miniconda@v2.1.1
+      with:
+        miniforge-variant: Mambaforge
+        miniforge-version: latest
+        use-mamba: true
+        mamba-version: "*"
+        channels: conda-forge,defaults
+        channel-priority: true
+        python-version: ${{ matrix.python-version }}
+        activate-environment: pudl-usage-metrics
+        environment-file: environment.yml
+    - shell: bash -l {0}
+      run: |
+        mamba info
+        mamba list
+        conda config --show-sources
+        conda config --show
+        printenv | sort
+
+    # Authentication via credentials json
+    - name: Authenticate gcloud
+      uses: "google-github-actions/auth@v0"
+      with:
+        credentials_json: "${{ secrets.GCE_SA_KEY }}"
```

Comment:
For dependabot updates to be able to run CI I think this needs to be stored in the dependabot secrets (but it would be accessed exactly the same way here).